### DISCUSSION TITLE
Docker Dependencies Not Grouped

### DISCUSSION THREAD

**Discussion by Kurt-von-Laven** [renovatebot/renovate#18814] [Request Help] [answered]
We have gotten significant value out of the `groupName` setting, which works well for us in most cases. However, it doesn't take effect in the case of our MegaLinter dependencies, [one of which has a datasource of docker](https://github.com/ScribeMD/.github/blob/f460658b81435dae3f9f1a29356ce62c7447af8f/default.json#L145-L154). Here is the relevant excerpt of the pertinent Dependency Dashboard (ScribeMD/pre-commit-hooks#132):

> ## Pending Approval
> 
> These branches will be created by Renovate only once you click their checkbox below.
> 
>  - [ ] chore(deps-dev): bump mega-linter-runner from v6.13.0 to v6.14.0
>  - [ ] chore(deps-dev): bump megalinter docker tag to v6.14.0
>  - [ ] 🔐 **Create all pending approval PRs at once** 🔐

[Here](https://github.com/ScribeMD/.github/blob/f460658b81435dae3f9f1a29356ce62c7447af8f/default.json#L38-L45) is the most relevant portion of our Renovate config. The comments in the raw Markdown suggest that mega-linter-runner has the expected branch name of `renovate/megalinter`, while the MegaLinter Docker tag instead has branch `renovate/megalinter-6.x`.

**Comment by viceice** [comment, marked as answer]
probably because the package rule matcher are case sensitive and your docker dep uses upper case letters

**Reply by Kurt-von-Laven** [reply]
To confirm, the intended behavior of `matchPackagePatterns` is to match the [`depName`](https://github.com/ScribeMD/.github/blob/f460658b81435dae3f9f1a29356ce62c7447af8f/default.json#L150) rather than the [`packageName`](https://github.com/ScribeMD/.github/blob/f460658b81435dae3f9f1a29356ce62c7447af8f/default.json#L151)?

**Reply by viceice** [reply]
yes, we like to change that in future, as packageName is the value used by datasources for lookup (deoName is only the fallback).

**Reply by Kurt-von-Laven** [reply]
Sounds like a good change; thank you for clearing this up. I'll make our regex match both the `depName` and `packageName` so that we aren't broken by this change in the future.

**Reply by Kurt-von-Laven** [reply]
I assume the same is true of `matchPackageNames`? In other words, `matchPackageNames` currently matches `depName` rather than `packageName`, but the plan is to change this in the future as well?

**Reply by rarkins** [reply]
Yes, in the next step it will match both, and warn if it matches only `depName`, and then in the final step it will match only against `packageName`
